app/Models/Asset.php

```diff
@@ -124,7 +125,14 @@ public function declinedCheckout(User $declinedBy, $signature)
         'requestable'      => 'nullable|boolean',
     ];
 
+    public function __construct(array $attributes = [])
+    {
+        parent::__construct($attributes);
+
+        $required_serial = Setting::getSettings()->required_serial;
 
+        $this->rules['serial'] = 'unique_undeleted:assets,serial|' . ($required_serial ? 'required' : 'nullable');
+    }
```

Comment:
Instead of messing with the model's constructor I think this would be better as a custom rule defined in a `Validator::extend` in `ValidationServiceProvider`. Does that make sense in this case?

Reply:
done, not as simple looking but feels like the correct way.